trustregistry/crud.py

```diff
@@ -51,26 +52,53 @@ def get_actor_by_id(db_session: Session, actor_id: str) -> db.Actor:
 
 def create_actor(db_session: Session, actor: Actor) -> db.Actor:
     bound_logger = logger.bind(body={"actor": actor})
-    bound_logger.info(
-        "Create actor in database. First assert actor ID does not already exist"
-    )
-    db_actor = db_session.query(db.Actor).filter(db.Actor.id == actor.id).one_or_none()
-
-    if db_actor:
-        bound_logger.info(
-            "Cannot create actor, as actor ID `{}` already exists in database.",
-            actor.id,
-        )
-        raise ActorAlreadyExistsException
-
-    bound_logger.debug("Adding actor to database")
-    db_actor = db.Actor(**actor.dict())
-    db_session.add(db_actor)
-    db_session.commit()
-    db_session.refresh(db_actor)
-
-    bound_logger.info("Successfully added actor to database.")
-    return db_actor
+    bound_logger.info("Try to create actor in database")
+
+    try:
+        bound_logger.debug("Adding actor to database")
+        db_actor = db.Actor(**actor.dict())
+        db_session.add(db_actor)
+        db_session.commit()
+        db_session.refresh(db_actor)
+
+        bound_logger.info("Successfully added actor to database.")
+        return db_actor
+
+    except IntegrityError as e:
+        db_session.rollback()
+        constraint_violation = str(e.orig).lower()
+
+        if "actors.id" in constraint_violation:
+            bound_logger.info(
+                "Bad request: An actor with ID already exists in database."
+            )
+            raise ActorAlreadyExistsException(
+                f"Bad request: An actor with ID: `{actor.id}` already exists in database."
+            )
+
+        elif "actors.name" in constraint_violation:
+            bound_logger.info(
+                "Bad request: An actor with name already exists in database."
+            )
+            raise ActorAlreadyExistsException(
+                f"Bad request: An actor with name: `{actor.name}` already exists in database."
+            )
+
+        elif "actors.did" in constraint_violation:
+            if "actors.didcomm_invitation" in constraint_violation:
+                bound_logger.info(
+                    "Bad request: An actor with DIDComm invitation already exists in database."
+                )
+                raise ActorAlreadyExistsException(
+                    "Bad request: An actor with DIDComm invitation already exists in database."
+                )
+            else:
+                bound_logger.info(
+                    "Bad request: An actor with DID already exists in database."
+                )
+                raise ActorAlreadyExistsException(
+                    f"Bad request: An actor with DID: `{actor.did}` already exists in database."
+                )
```

Comment:
`elif` has no final `else` case - so could swallow `IntegrityError` that doesn't match on checks. Should be one final else to raise a generic "Unique constraint failed" message, in case specific if checks don't match.

I think the nested one:
```py
       elif "actors.did" in constraint_violation:
            if "actors.didcomm_invitation" in constraint_violation:
```

can just be replaced with `elif "actors.didcomm_invitation" in ... :` followed by `elif "actors.did" in ...` and then else: Generic log/response (with an error log to indicate that a constraint failed which isn't manually handled)